runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -401,9 +416,34 @@ public void hideWidgets(WidgetHiddenChanged event)
 			return;
 		}
 
+		if (config.enableDateInFilename())
+		{
+			fileName = TIME_FORMAT.format(new Date() + "-" + fileName);
+		}
+
 		takeScreenshot(fileName);
 	}
 
+	@Subscribe
+	public void onGameStatechange(GameStateChanged gameStateChanged)
+	{
+		if (worldResult == null)
+		{
+			return;
+		}
+
+		final World world = worldResult.findWorld(client.getWorld());
```

Comment:
You dont need to do this, there is client.getWorldTypes

Reply:
Good point, removed the dependency on `WorldResult` and switched to using `client.getWorldType` on game state change. :+1: